### PR TITLE
Try running tests in GitHub Actions

### LINKED ISSUE #676: Travis taking forever to finish..move to git-actions?
We moved plots2 to git-actions and the CI is now running seamless...we could attempt to do the same here...
 plots2 PR: https://github.com/publiclab/plots2/pull/8795/

### PR DESCRIPTION
**Pull request by jywarren**
Fixes #676

### CODE REVIEW

**Comment by jywarren** [comment]
PL undefined error from main branch! https://travis-ci.com/github/publiclab/PublicLab.Editor/jobs/399630193

**Comment by jywarren** [comment]
It's been the case since https://github.com/publiclab/PublicLab.Editor/pull/541, where we added `--force` to `grunt jasmine` - it didn't fail there but at some point later it did and then we stopped being able to see when jasmine failed...

**Comment by jywarren** [comment]
Here is where it began to fail!!!! https://github.com/publiclab/PublicLab.Editor/pull/597

**Comment by jywarren** [comment]
I'll try pulling in each commit of `woofmark` from this list to see when it starts to break: 

https://github.com/jywarren/woofmark/commits/plots2

whoaaaaa. i think we need to go back to https://github.com/jywarren/woofmark/compare/12e2c1d4559268e2de605ad6084d5dce24ea42c6...39237b798308e6855acde1a938c18a8e4da62fce ... yikes...

**Comment by jywarren** [comment]
Reverting back to 12e2c1d or 3695d655c7d11fd3cee8e19f534ccc6722f1297f didn't work... :-(((

**Comment by jywarren** [comment]
Now trying to look through every line of this commit which is when tests started to show this error: https://github.com/jywarren/woofmark/commit/5cc44d66ed3880d6089954a995eafdfc7b9aede9

**Comment by jywarren** [comment]
Couldn't find anything! But, i did see that `bureaucracy` also changed pinned version: https://github.com/publiclab/PublicLab.Editor/pull/597/files

**Comment by jywarren** [comment]
Seems unlikely to be the problem - the change was really minimal: https://github.com/bevacqua/bureaucracy/compare/master...jywarren:xhrOptions

**Comment by jywarren** [comment]
So whatever happened in https://github.com/publiclab/PublicLab.Editor/pull/597, reverting it doesn't get past the `PL not defined` error. 

My theory also that something is breaking in our dist file doesn't make sense either... it's working in Jest. Maybe it's ES6 related? But, then why would... gah... that's back to the status we had in #614.

**Comment by jywarren** [comment]
Even that's not quite right - we should be Chrome Headless and not Phantomjs now, right? https://github.com/gruntjs/grunt-contrib-jasmine/issues/252#issuecomment-391732287

**Comment by jywarren** [comment]
Ah, that only happens in `grunt-contrib-jasmine` v2+ -- we need to upgrade

**Comment by jywarren** [comment]
And noting we gave up on that upgrade in https://github.com/publiclab/PublicLab.Editor/pull/168

**Comment by jywarren** [comment]
Haha! `log: Access to XMLHttpRequest at 'file:///home/runner/work/PublicLab.Editor/PublicLab.Editor/spec/javascripts/fixtures/index.html?_=1611708375140' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, https.`

I'll rework that to use a server.

**Comment by jywarren** [comment]
Ooohhh!!!!

```
Error: Jasmine Ajax was unable to install over a custom XMLHttpRequest. Is Jasmine Ajax already installed? (1)
>> Error: Jasmine Ajax was unable to install over a custom XMLHttpRequest. Is Jasmine Ajax already installed? Error: Jasmine Ajax was unable to install over a custom XMLHttpRequest. Is Jasmine Ajax already installed?
>>     at MockAjax.install (file:///workspace/PublicLab.Editor/node_modules/jasmine-ajax/lib/mock-ajax.js:575:15)
>>     at UserContext.<anonymous> (file:///workspace/PublicLab.Editor/spec/javascripts/main_image_module_spec.js:50:18)
>>     at <Jasmine>
```

```
   X sends AJAX request on editor.publish()
     TypeError: Cannot read property 'blurredLocation' of undefined (1)
>> TypeError: Cannot read property 'blurredLocation' of undefined TypeError: Cannot read property 'blurredLocation' of undefined
>>     at Class.publish (file:///workspace/PublicLab.Editor/dist/PublicLab.Editor.js:21683:34)
>>     at UserContext.<anonymous> (file:///workspace/PublicLab.Editor/spec/javascripts/editor_spec.js:60:12)
>>     at <Jasmine>
   X exists, and has a textarea
     Expected undefined not to be undefined. (1)
>> Expected undefined not to be undefined. Error: Expected undefined not to be undefined.
>>     at <Jasmine>
>>     at UserContext.<anonymous> (file:///workspace/PublicLab.Editor/spec/javascripts/editor_spec.js:14:39)
>>     at <Jasmine>
     Expected <textarea class="ple-textarea form-control wk-hide">...</textarea> to be undefined. (2)
>> Expected <textarea class="ple-textarea form-control wk-hide">...</textarea> to be undefined. Error: Expected <textarea class="ple-textarea form-control wk-hide">...</textarea> to be undefined.
>>     at <Jasmine>
>>     at UserContext.<anonymous> (file:///workspace/PublicLab.Editor/spec/javascripts/editor_spec.js:17:37)
>>     at <Jasmine>
```

are these all ajax errors???

**Comment by jywarren** [comment]
I tried setting `'allow-sync-xhr-in-page-dismissal']` in gruntfile because we're getting synchronous AJAX errors, but it didn't seem to work:

```
log: Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
```

**Comment by jywarren** [comment]
Noting synch ajax was disallowed in chrome 80, should check the chrome driver version and how to flag that one

**Comment by jywarren** [comment]
Could use StubRequest to switch jquery-ajax to synchronous: https://jasmine.github.io/2.0/ajax.html#section-12

https://github.com/jasmine/jasmine-ajax/issues/26

**Comment by jywarren** [comment]
Wow this is working! I think we should merge it. However, i do see this error sometimes: `log: Failed to load resource: net::ERR_FILE_NOT_FOUND` -- it could be for a vendor asset we aren't using?

File: Gruntfile.js [hunk 1]
```diff
@@ -43,8 +43,13 @@ module.exports = function(grunt) {
 
         jasmine: {
           publiclabeditor: {
-            src: 'dist/*.js',
+            src: 'dist/PublicLab.Editor.js',
             options: {
+              allowFileAccess: true,
+              sandboxArgs: {
+                args: ['--no-sandbox','allow-sync-xhr-in-page-dismissal'],
```
Comment (jywarren) [inline Gruntfile.js:50]:
```suggestion
                args: ['--no-sandbox', '--allow-sync-xhr-in-page-dismissal'],
```

**Comment by jywarren** [comment]
Yayyyyy!!!

**Comment by cesswairimu** [comment]
🎉🎉🎉🎉🎉

**Pull request merged**